### PR TITLE
Initial bloom/angular monorepo rewrite skeleton [DAH-691][DAH-727]

### PR DESCRIPTION
**Pull request by james-wills-sf**
Feasibility testing: DAH-691
Rewrite starting point: DAH-727

Create a starting point for the bloom rewrite code.

### CODE REVIEW

**Comment by james-wills-sf** [comment]
@akegan it looks like circleCI is still having issues, but this change is ready for a qa/first pass code review.

File: app/controllers/home_controller.rb
Comment (akegan) [inline app/controllers/home_controller.rb:6]: I tested this to see if it causes a problem, but it looks like we still redirect as expected if you go to /mohcd-plus-housing

File: app/views/layouts/application-react.html.slim
Comment (akegan) [inline app/views/layouts/application-react.html.slim:35]: never new that was an option! is that something that we have on angular too?

File: bookmarklet.js
Comment (akegan) [inline bookmarklet.js:5]: weird. no idea what this is. I guess some firefox extension?

File: config/initializers/new_framework_defaults.rb
Comment (akegan) [inline config/initializers/new_framework_defaults.rb:22]: was this change and the one above necessary? both of the settings that were here seem reasonable?

File: config/webpack/development.js
Comment (akegan) [inline config/webpack/development.js:1]: at some point I might ask what the actual differences are between environments

File: config/webpacker.yml
Comment (akegan) [inline config/webpacker.yml:33]: not a big deal, but wondering where this list came from. I have never heard of .woff files!

File: spec/controllers/overrides/registrations_controller_spec.rb
Comment (akegan) [inline spec/controllers/overrides/registrations_controller_spec.rb:61]: wondering what the id is for jack2doe and how the update finds the id?/user?

File: spec/e2e/pages/short-form/you-name.coffee
Comment (akegan) [inline spec/e2e/pages/short-form/you-name.coffee:5]: what does the super() do in these?

File: spec/javascripts/fixtures/json/listings-api-ami.json
Comment (akegan) [inline spec/javascripts/fixtures/json/listings-api-ami.json:1]: did anything change in here? wondering why this was reformatted

**Review by akegan — CHANGES_REQUESTED** [review]
It looks like the package size thing is OK. Same size as on prod unless you load the actual react page
From incognito load of the angular home page
image

vs for prod
image
seems about the same/slightly less than prod actually. 

Here is the react homepage 
image



Only bug that I found that is a blocker is that if you set the ENV var to false so the default home page is angular, the link back to the home page still doesn't respect the language you're on. So if you're on https://dahlia-web-full-pr-1436.herokuapp.com/es/listings/for-sale and then click the logo, you get sent to the english home page

File: config/initializers/new_framework_defaults.rb
Comment (software-project) [reply config/initializers/new_framework_defaults.rb:22]: @akegan both of those options were removed from Rails. The first one actually didnt have any effect since Rails 5.1. The second one was replaced by explicitly calling `throw :abort` in Rails controller.

File: bookmarklet.js
Comment (software-project) [reply bookmarklet.js:5]: I even wonder what that was even supposed to do. It wasnt referenced anywhere, even on the initial commit that added it.

Comment (akegan) [reply bookmarklet.js:5]: strange! good to get rid of it then

File: config/initializers/new_framework_defaults.rb
Comment (akegan) [reply config/initializers/new_framework_defaults.rb:22]: thanks!

File: config/webpack/development.js
Comment (software-project) [reply config/webpack/development.js:1]: At this point, no difference at all. But I suppose for production we will want to compress and uglify css as it's 3mb in development ;)

File: config/webpacker.yml
Comment (software-project) [reply config/webpacker.yml:33]: This is autogenerated file when you install webpacker. And woff is a font format for dog lovers ;)

File: spec/controllers/overrides/registrations_controller_spec.rb
Comment (james-wills-sf) [reply spec/controllers/overrides/registrations_controller_spec.rb:61]: if you look at the [update method on the base class](https://github.com/lynndylanhurley/devise_token_auth/blob/master/app/controllers/devise_token_auth/registrations_controller.rb#L69) it isn't querying by id, it's applying update changes to the existing @resource, which we assigned to the user in the `before` block.

File: spec/e2e/pages/short-form/you-name.coffee
Comment (james-wills-sf) [reply spec/e2e/pages/short-form/you-name.coffee:5]: I just tried to remove it and our e2e tests failed, I think it's because we've updated our ES version and now it's a requirement: https://github.com/jashkenas/coffeescript/issues/4547

File: spec/javascripts/fixtures/json/listings-api-ami.json
Comment (james-wills-sf) [reply spec/javascripts/fixtures/json/listings-api-ami.json]: i don't think any of this should have changed. reverted it

File: config/webpacker.yml
Comment (akegan) [reply config/webpacker.yml:33]: 😆  thanks!

**Comment by james-wills-sf** [comment]
@akegan good catch on the language thing. I think it's going to take a little longer than this afternoon to fix. It's probably very related to DAH-739, since if we can handle all routing in one place, that's also where we can handle routing the language url path

**Review by akegan — APPROVED** [review]
After talking to James, I am OK with merging this as is and following up quickly w/ the language-missing-in-redirect-issue

**Comment by james-wills-sf** [comment]
@akegan I just updated this with better behavior for translations, and I'll make a ticket about the 404 issue. The fix is slightly hacky, especially in routes.rb, but I think on the angular side it's actually a better system to do the redirects in stateChangeStart, so we can catch it all in one place.

I'm not going to rush and merge this tonight, maybe if you have time you can look at the new commit and re-qa on monday and if @software-project feels it will speed him up he can merge? Otherwise I'll merge first thing tuesday.

**Comment by akegan** [comment]
OK QA looks good to me so up to @software-project if you want to merge before tuesday morning 😄

**Review by software-project — APPROVED** [review]
Looks good. Really nice job @james-wills-sf on the details.

**Pull request merged**
